src/couch_mrview/src/couch_mrview_index.erl

```diff
@@ -122,7 +122,7 @@ open(Db, State0) ->
 
     case couch_mrview_util:open_file(IndexFName) of
         {ok, Fd} ->
-            case (catch couch_file:read_header(Fd)) of
+            case couch_file:read_header(Fd) of
```

Comment:
So what happens when there's an error? I definitely think we should only trigger the index reset upon `no_valid_header`, but dropping the `catch` changes behavior to where errors can now bubble up out of this function, which may or may not be a good idea.